sonic/eth-tests.jenkinsfile

```
@@ -0,0 +1,57 @@
+pipeline {
+    agent { label 'x86-4-32-s' }
+
+    options {
+        timestamps ()
+        timeout (time: 4, unit: 'HOURS') // ~2.5h
+        disableConcurrentBuilds(abortPrevious: false)
+    }
+
+    environment {
+        TMPDB = '/mnt/tmp-disk'
+    }
+
+     parameters {
+        string(name: 'SonicVersion', defaultValue: "develop", description: 'Can be either branch name or commit hash.')
+    }
+
+    stages {
+        stage('Checkout') {
+            steps {
+                script {
+                    currentBuild.description = "Building on ${env.NODE_NAME}"
+                }
+
+                checkout scmGit(
+                    branches: [[name: "${SonicVersion}"]],
+                    userRemoteConfigs: [[url: 'https://github.com/Fantom-foundation/Sonic.git']]
+                )
+
+                dir('tests/test-data') {
+                    checkout scmGit(
+                        branches: [[name: "develop"]],
+                        userRemoteConfigs: [[url: 'https://github.com/ethereum/tests.git']]
+                    )
+                }
+            }
+        }
+
+        stage('Build') {
+            steps {
+                sh "go test ./tests -timeout 4h"  
```

```suggestion
                sh "go test ./tests"  
```
Timeout for the command is the same value as for the whole pipeline. This is a bit redundant I'd say as the job will be killed sooner by jenkins every time because the preceding stage take some time too.

---

I updated to 2hours. Removing the timeout would mean 10mins only and it is not enough for this this.

---

Understood :)